### ISSUE TITLE
waiting for selector (timeouterror)

### ISSUE THREAD

**Issue by haupt-pascal** [fawazahmed0/youtube-uploader#220]
I am receiving the following error - is this my fault or are you able to fix this soon? 

TimeoutError: waiting for selector `input[type="password"]:not([aria-hidden="true"])` failed: timeout 60000ms exceeded
scheduler       |         at new WaitTask (/home/node/app/node_modules/puppeteer/src/common/DOMWorld.ts:938:28)
scheduler       |         at DOMWorld._waitForSelectorInPage (/home/node/app/node_modules/puppeteer/src/common/DOMWorld.ts:772:22)
scheduler       |         at Object.internalHandler.waitFor (/home/node/app/node_modules/puppeteer/src/common/QueryHandler.ts:82:23)
scheduler       |         at DOMWorld.waitForSelector (/home/node/app/node_modules/puppeteer/src/common/DOMWorld.ts:611:25)
scheduler       |         at Frame.waitForSelector (/home/node/app/node_modules/puppeteer/src/common/FrameManager.ts:1367:47)
scheduler       |         at Page.waitForSelector (/home/node/app/node_modules/puppeteer/src/common/Page.ts:3336:29)
scheduler       |         at login (/home/node/app/node_modules/youtube-videos-uploader/src/upload.ts:1140:25)
scheduler       |         at processTicksAndRejections (node:internal/process/task_queues:95:5)
scheduler       |         at loadAccount (/home/node/app/node_modules/youtube-videos-uploader/src/upload.ts:954:13)
scheduler       |         at upload (/home/node/app/node_modules/youtube-videos-uploader/src/upload.ts:70:9)
scheduler       | [16:27:09.349] WARN (adonis_app/1): pino.final with prettyPrint does not support flushing
scheduler       | /home/node/app/node_modules/puppeteer/lib/cjs/puppeteer/common/DOMWorld.js:680
scheduler       |             const timeoutError = new Errors_js_1.TimeoutError(`waiting for ${options.title} failed: timeout ${options.timeout}ms exceeded`);
scheduler       |                                  ^
scheduler       | 
scheduler       | TimeoutError: waiting for selector `input[type="password"]:not([aria-hidden="true"])` failed: timeout 60000ms exceeded
scheduler       |     at new WaitTask (/home/node/app/node_modules/puppeteer/src/common/DOMWorld.ts:938:28)
scheduler       |     at DOMWorld._waitForSelectorInPage (/home/node/app/node_modules/puppeteer/src/common/DOMWorld.ts:772:22)
scheduler       |     at Object.internalHandler.waitFor (/home/node/app/node_modules/puppeteer/src/common/QueryHandler.ts:82:23)
scheduler       |     at DOMWorld.waitForSelector (/home/node/app/node_modules/puppeteer/src/common/DOMWorld.ts:611:25)
scheduler       |     at Frame.waitForSelector (/home/node/app/node_modules/puppeteer/src/common/FrameManager.ts:1367:47)
scheduler       |     at Page.waitForSelector (/home/node/app/node_modules/puppeteer/src/common/Page.ts:3336:29)
scheduler       |     at login (/home/node/app/node_modules/youtube-videos-uploader/src/upload.ts:1140:25)
scheduler       |     at processTicksAndRejections (node:internal/process/task_queues:95:5)
scheduler       |     at loadAccount (/home/node/app/node_modules/youtube-videos-uploader/src/upload.ts:954:13)
scheduler       |     at upload (/home/node/app/node_modules/youtube-videos-uploader/src/upload.ts:70:9)
scheduler       | 
scheduler       | Node.js v18.16.0

**Comment by N0itx** [comment]
same

**Comment by haupt-pascal** [comment]
This is part of a puppeteer-extra bug, I will link the issue from me for you. Just that you guys can follow it: 

https://github.com/berstend/puppeteer-extra/issues/822#issuecomment-1685656287

**Comment by jamesjiayu** [comment]
same. 
thank you for https://github.com/berstend/puppeteer-extra/issues/822#issuecomment-1685656287. but it's not fixed, right? so this is not fixed too...

**Comment by jamesjiayu** [comment]
funny thing, after restarting my ubantu 22 lts, the error s gone. but new error: 
/home/j/Desktop/youtubeUploader/node_modules/puppeteer/lib/cjs/puppeteer/common/FrameManager.js:195
                    ? new Error(`${response.errorText} at ${url}`)
                      ^

Error: net::ERR_ABORTED at https://www.youtube.com/upload?persist_gl=1&gl=US&persist_hl=1&hl=en
    at navigate (/home/j/Desktop/youtubeUploader/node_modules/puppeteer/lib/cjs/puppeteer/common/FrameManager.js:195:23)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async FrameManager.navigateFrame (/home/j/Desktop/youtubeUploader/node_modules/puppeteer/lib/cjs/puppeteer/common/FrameManager.js:171:21)
    at async Frame.goto (/home/j/Desktop/youtubeUploader/node_modules/puppeteer/lib/cjs/puppeteer/common/FrameManager.js:584:16)
    at async Page.goto (/home/j/Desktop/youtubeUploader/node_modules/puppeteer/lib/cjs/puppeteer/common/Page.js:1109:16)
    at async login (/home/j/Desktop/youtubeUploader/node_modules/youtube-videos-uploader/dist/upload.js:888:5)
    at async loadAccount (/home/j/Desktop/youtubeUploader/node_modules/youtube-videos-uploader/dist/upload.js:782:13)
    at async upload (/home/j/Desktop/youtubeUploader/node_modules/youtube-videos-uploader/dist/upload.js:46:9)

Node.js v18.17.1

**Comment by karimb** [comment]
My guess is that puppeteer is waiting for a user interaction and times out. Deactivate the headless mode to solve it:
Instead of
`upload (credentials, [video1, video2]).then(console.log)`

do:
`upload (credentials, [video1, video2], **{headless:false}**).then(console.log)`

**Comment by ali-kingX** [comment]
@karimb Hey i did what you did, I get a new error:
TypeError: text is not iterable
    at Keyboard.type (C:\Users\laure\Desktop\Video Generator\node_modules\puppeteer\lib\cjs\puppeteer\common\Input.js:195:28)
    at ElementHandle.type (C:\Users\laure\Desktop\Video Generator\node_modules\puppeteer\lib\cjs\puppeteer\common\JSHandle.js:739:79)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async DOMWorld.type (C:\Users\laure\Desktop\Video Generator\node_modules\puppeteer\lib\cjs\puppeteer\common\DOMWorld.js:487:9)
    at async login (C:\Users\laure\Desktop\Video Generator\node_modules\youtube-videos-uploader\dist\upload.js:976:9)
    at async loadAccount (C:\Users\laure\Desktop\Video Generator\node_modules\youtube-videos-uploader\dist\upload.js:838:13)
    at async upload (C:\Users\laure\Desktop\Video Generator\node_modules\youtube-videos-uploader\dist\upload.js:52:9)
